extract/extract.py

```diff
@@ -2179,6 +2187,20 @@ def strip_overlay(self, request, file_path):
         shutil.move(out_path, os.path.join(self.working_directory, sha256hash))
         return (os.path.join(self.working_directory, sha256hash), None, None)
 
+    def extract_pyc(self, filepath):
```

Comment:
Not sure if you want to change the name, but technically this method is taking a path to an already extracted pyc file and then attempting to decomplie. It's not doing the pyc extraction.

Did you also want to include doc string?

    """Attempt to decompile the pyc file at the given filepath.

    Successfully decomplied files will be written back to the `working_directory`.

    Args:
        filepath: path to pyc file

    Returns:
        The filepath to the decompiled script.
    """

Reply:
Thank you for the docstring, I just added it.
Regarding the name, most of the functions are in the format of `extract_<original_file>`, like `extract_pdf_passwordprotected`, `extract_vbe`, `extract_zip`, `extract_xxe`, `extract_office`. I could rename it `decompile_pyc()`, but I think it fits the pattern that is most repeated in the module.